<<<<<<< HEAD
From 38431784171b52a0060cf330a52b2bd83fb92c45 Mon Sep 17 00:00:00 2001
From: Bing Jiang <bing.jiang@amlogic.com>
Date: Mon, 2 Nov 2020 16:02:04 +0800
Subject: [PATCH 1/3] uboot: update root system config [1/1]

PD#SWPL-35835

Problem:
rdk support absystem

Solution:
update root system config

Verify:
S4_AQ222 S4_AP222 S4_AP223 SC2_AH212 SC2_AH219 SC2_AH232

Change-Id: I71d09b8a425f151aba55ea83b46a19540104dfbd
Signed-off-by: Bing Jiang <bing.jiang@amlogic.com>
---
 board/amlogic/configs/s4_ap222.h             | 9 +++++++++
 board/amlogic/configs/s4_ap223.h             | 9 +++++++++
 board/amlogic/configs/s4_aq222.h             | 9 +++++++++
 board/amlogic/configs/sc2_ah212.h            | 9 +++++++++
 board/amlogic/configs/sc2_ah219.h            | 9 +++++++++
 board/amlogic/configs/sc2_ah232.h            | 9 +++++++++
 board/amlogic/defconfigs/s4_ap222_defconfig  | 2 +-
 board/amlogic/defconfigs/s4_ap223_defconfig  | 2 +-
 board/amlogic/defconfigs/s4_aq222_defconfig  | 2 +-
 board/amlogic/defconfigs/sc2_ah212_defconfig | 2 +-
 board/amlogic/defconfigs/sc2_ah219_defconfig | 2 +-
 board/amlogic/defconfigs/sc2_ah232_defconfig | 2 +-
 12 files changed, 60 insertions(+), 6 deletions(-)

diff --git a/board/amlogic/configs/s4_ap222.h b/board/amlogic/configs/s4_ap222.h
index 5bf8b3c50a..381febdef6 100644
--- a/board/amlogic/configs/s4_ap222.h
+++ b/board/amlogic/configs/s4_ap222.h
=======
Index: v2019/board/amlogic/configs/s4_ap222.h
===================================================================
--- v2019.orig/board/amlogic/configs/s4_ap222.h
+++ v2019/board/amlogic/configs/s4_ap222.h
>>>>>>> 335d8f964fd94ca0056eec8f6c98c4ad17d21364
@@ -178,6 +178,15 @@
                 "if test ${active_slot} != normal; then "\
                     "setenv bootargs ${bootargs} androidboot.slot_suffix=${active_slot};"\
                 "fi;"\
+                "if test ${avb2} = 0; then "\
+                    "if test ${active_slot} = _a; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system_a;"\
+                    "else if test ${active_slot} = _b; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system_b;"\
+                    "else if test ${active_slot} = normal; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system;"\
+                    "fi;fi;fi;"\
+                "fi;"\
                 "if fdt addr ${dtb_mem_addr}; then else echo retry common dtb; run common_dtb_load; fi;"\
                 "setenv loadaddr ${loadaddr_kernel};"\
                 "if imgread kernel ${boot_part} ${loadaddr}; then bootm ${loadaddr}; fi;"\
<<<<<<< HEAD
diff --git a/board/amlogic/configs/s4_ap223.h b/board/amlogic/configs/s4_ap223.h
index 9065b1871f..da69fbcbc9 100644
--- a/board/amlogic/configs/s4_ap223.h
+++ b/board/amlogic/configs/s4_ap223.h
=======
Index: v2019/board/amlogic/configs/s4_ap223.h
===================================================================
--- v2019.orig/board/amlogic/configs/s4_ap223.h
+++ v2019/board/amlogic/configs/s4_ap223.h
>>>>>>> 335d8f964fd94ca0056eec8f6c98c4ad17d21364
@@ -178,6 +178,15 @@
                 "if test ${active_slot} != normal; then "\
                     "setenv bootargs ${bootargs} androidboot.slot_suffix=${active_slot};"\
                 "fi;"\
+                "if test ${avb2} = 0; then "\
+                    "if test ${active_slot} = _a; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system_a;"\
+                    "else if test ${active_slot} = _b; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system_b;"\
+                    "else if test ${active_slot} = normal; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system;"\
+                    "fi;fi;fi;"\
+                "fi;"\
                 "if fdt addr ${dtb_mem_addr}; then else echo retry common dtb; run common_dtb_load; fi;"\
                 "setenv loadaddr ${loadaddr_kernel};"\
                 "if imgread kernel ${boot_part} ${loadaddr}; then bootm ${loadaddr}; fi;"\
<<<<<<< HEAD
diff --git a/board/amlogic/configs/s4_aq222.h b/board/amlogic/configs/s4_aq222.h
index 8cf8925af0..f05f97e5c7 100644
--- a/board/amlogic/configs/s4_aq222.h
+++ b/board/amlogic/configs/s4_aq222.h
=======
Index: v2019/board/amlogic/configs/s4_aq222.h
===================================================================
--- v2019.orig/board/amlogic/configs/s4_aq222.h
+++ v2019/board/amlogic/configs/s4_aq222.h
>>>>>>> 335d8f964fd94ca0056eec8f6c98c4ad17d21364
@@ -168,6 +168,15 @@
                 "if test ${active_slot} != normal; then "\
                     "setenv bootargs ${bootargs} androidboot.slot_suffix=${active_slot};"\
                 "fi;"\
+                "if test ${avb2} = 0; then "\
+                    "if test ${active_slot} = _a; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system_a;"\
+                    "else if test ${active_slot} = _b; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system_b;"\
+                    "else if test ${active_slot} = normal; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system;"\
+                    "fi;fi;fi;"\
+                "fi;"\
                 "if fdt addr ${dtb_mem_addr}; then else echo retry common dtb; run common_dtb_load; fi;"\
                 "setenv loadaddr ${loadaddr_kernel};"\
                 "if imgread kernel ${boot_part} ${loadaddr}; then bootm ${loadaddr}; fi;"\
<<<<<<< HEAD
diff --git a/board/amlogic/configs/sc2_ah212.h b/board/amlogic/configs/sc2_ah212.h
index 3d169a8ed6..0dc0ab88fc 100644
--- a/board/amlogic/configs/sc2_ah212.h
+++ b/board/amlogic/configs/sc2_ah212.h
@@ -177,6 +177,15 @@
                 "if test ${active_slot} != normal; then "\
                     "setenv bootargs ${bootargs} androidboot.slot_suffix=${active_slot};"\
                 "fi;"\
+                "if test ${avb2} = 0; then "\
=======
Index: v2019/board/amlogic/configs/sc2_ah212.h
===================================================================
--- v2019.orig/board/amlogic/configs/sc2_ah212.h
+++ v2019/board/amlogic/configs/sc2_ah212.h
@@ -178,6 +178,15 @@
                 "if test ${active_slot} != normal; then "\
                     "setenv bootargs ${bootargs} androidboot.slot_suffix=${active_slot};"\
                 "fi;"\
+                "if test ${avb2} = 1; then "\
>>>>>>> 335d8f964fd94ca0056eec8f6c98c4ad17d21364
+                    "if test ${active_slot} = _a; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system_a;"\
+                    "else if test ${active_slot} = _b; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system_b;"\
+                    "else if test ${active_slot} = normal; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system;"\
+                    "fi;fi;fi;"\
+                "fi;"\
                 "if fdt addr ${dtb_mem_addr}; then else echo retry common dtb; run common_dtb_load; fi;"\
                 "setenv loadaddr ${loadaddr_kernel};"\
                 "if imgread kernel ${boot_part} ${loadaddr}; then bootm ${loadaddr}; fi;"\
Index: v2019/board/amlogic/configs/sc2_ah219.h
===================================================================
--- v2019.orig/board/amlogic/configs/sc2_ah219.h
+++ v2019/board/amlogic/configs/sc2_ah219.h
@@ -170,6 +170,15 @@
                 "if test ${active_slot} != normal; then "\
                     "setenv bootargs ${bootargs} androidboot.slot_suffix=${active_slot};"\
                 "fi;"\
+                "if test ${avb2} = 0; then "\
+                    "if test ${active_slot} = _a; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system_a;"\
+                    "else if test ${active_slot} = _b; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system_b;"\
+                    "else if test ${active_slot} = normal; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system;"\
+                    "fi;fi;fi;"\
+                "fi;"\
                 "if fdt addr ${dtb_mem_addr}; then else echo retry common dtb; run common_dtb_load; fi;"\
                 "setenv loadaddr ${loadaddr_kernel};"\
                 "if imgread kernel ${boot_part} ${loadaddr}; then bootm ${loadaddr}; fi;"\
Index: v2019/board/amlogic/configs/sc2_ah232.h
===================================================================
--- v2019.orig/board/amlogic/configs/sc2_ah232.h
+++ v2019/board/amlogic/configs/sc2_ah232.h
@@ -183,6 +183,15 @@
                 "if test ${active_slot} != normal; then "\
                     "setenv bootargs ${bootargs} androidboot.slot_suffix=${active_slot};"\
                 "fi;"\
+                "if test ${avb2} = 0; then "\
+                    "if test ${active_slot} = _a; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system_a;"\
+                    "else if test ${active_slot} = _b; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system_b;"\
+                    "else if test ${active_slot} = normal; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system;"\
+                    "fi;fi;fi;"\
+                "fi;"\
                 "if fdt addr ${dtb_mem_addr}; then else echo retry common dtb; run common_dtb_load; fi;"\
                 "setenv loadaddr ${loadaddr_kernel};"\
                 "if imgread kernel ${boot_part} ${loadaddr}; then bootm ${loadaddr}; fi;"\
Index: v2019/board/amlogic/defconfigs/s4_ap222_defconfig
===================================================================
--- v2019.orig/board/amlogic/defconfigs/s4_ap222_defconfig
+++ v2019/board/amlogic/defconfigs/s4_ap222_defconfig
@@ -36,7 +36,7 @@ CONFIG_CMD_BMP=y
 CONFIG_CMD_CACHE=y
 # CONFIG_CMD_MISC is not set
 CONFIG_CMD_MTDPARTS=y
-CONFIG_CMD_BOOTCTOL_AVB=y
+# CONFIG_CMD_BOOTCTOL_AVB=y
 CONFIG_CMD_WATCHDOG=y
 CONFIG_CMD_JTAG=y
 CONFIG_CMD_INI=y
<<<<<<< HEAD
diff --git a/board/amlogic/defconfigs/s4_ap223_defconfig b/board/amlogic/defconfigs/s4_ap223_defconfig
index 551c09b6a3..50ecb46059 100644
--- a/board/amlogic/defconfigs/s4_ap223_defconfig
+++ b/board/amlogic/defconfigs/s4_ap223_defconfig
=======
Index: v2019/board/amlogic/defconfigs/s4_ap223_defconfig
===================================================================
--- v2019.orig/board/amlogic/defconfigs/s4_ap223_defconfig
+++ v2019/board/amlogic/defconfigs/s4_ap223_defconfig
>>>>>>> 335d8f964fd94ca0056eec8f6c98c4ad17d21364
@@ -35,7 +35,7 @@ CONFIG_CMD_BMP=y
 CONFIG_CMD_CACHE=y
 # CONFIG_CMD_MISC is not set
 CONFIG_CMD_MTDPARTS=y
-CONFIG_CMD_BOOTCTOL_AVB=y
+# CONFIG_CMD_BOOTCTOL_AVB=y
 CONFIG_CMD_WATCHDOG=y
 CONFIG_CMD_JTAG=y
 CONFIG_CMD_INI=y
<<<<<<< HEAD
diff --git a/board/amlogic/defconfigs/s4_aq222_defconfig b/board/amlogic/defconfigs/s4_aq222_defconfig
index 10a013dcfa..045787fc6d 100644
--- a/board/amlogic/defconfigs/s4_aq222_defconfig
+++ b/board/amlogic/defconfigs/s4_aq222_defconfig
=======
Index: v2019/board/amlogic/defconfigs/s4_aq222_defconfig
===================================================================
--- v2019.orig/board/amlogic/defconfigs/s4_aq222_defconfig
+++ v2019/board/amlogic/defconfigs/s4_aq222_defconfig
>>>>>>> 335d8f964fd94ca0056eec8f6c98c4ad17d21364
@@ -36,7 +36,7 @@ CONFIG_CMD_BMP=y
 CONFIG_CMD_CACHE=y
 # CONFIG_CMD_MISC is not set
 CONFIG_CMD_MTDPARTS=y
-CONFIG_CMD_BOOTCTOL_AVB=y
+# CONFIG_CMD_BOOTCTOL_AVB=y
 CONFIG_CMD_WATCHDOG=y
 CONFIG_CMD_JTAG=y
 CONFIG_AML_PARTITION=y
Index: v2019/board/amlogic/defconfigs/sc2_ah219_defconfig
===================================================================
--- v2019.orig/board/amlogic/defconfigs/sc2_ah219_defconfig
+++ v2019/board/amlogic/defconfigs/sc2_ah219_defconfig
@@ -36,7 +36,7 @@ CONFIG_CMD_BMP=y
 CONFIG_CMD_CACHE=y
 # CONFIG_CMD_MISC is not set
 CONFIG_CMD_MTDPARTS=y
-CONFIG_CMD_BOOTCTOL_AVB=y
+# CONFIG_CMD_BOOTCTOL_AVB=y
 CONFIG_CMD_WATCHDOG=y
 CONFIG_CMD_JTAG=y
 CONFIG_CMD_INI=y
Index: v2019/board/amlogic/defconfigs/sc2_ah232_defconfig
===================================================================
--- v2019.orig/board/amlogic/defconfigs/sc2_ah232_defconfig
+++ v2019/board/amlogic/defconfigs/sc2_ah232_defconfig
@@ -36,7 +36,7 @@ CONFIG_CMD_IR=y
 CONFIG_CMD_BMP=y
 CONFIG_CMD_CACHE=y
 # CONFIG_CMD_MISC is not set
-CONFIG_CMD_BOOTCTOL_AVB=y
+# CONFIG_CMD_BOOTCTOL_AVB=y
 CONFIG_CMD_WATCHDOG=y
 CONFIG_CMD_JTAG=y
 CONFIG_CMD_INI=y
